sdks/java/core/src/test/java/org/apache/beam/sdk/schemas/transforms/RenameFieldsTest.java

```diff
@@ -258,4 +265,65 @@ public void renameNestedInMapFields() {
     PAssert.that(renamed).containsInAnyOrder(expectedRows);
     pipeline.run();
   }
+
+  @Test
+  public void testRenameRow() {
+    Schema nestedSchema = Schema.builder().addStringField("field1").addInt32Field("field2").build();
+    Schema schema =
+        Schema.builder().addStringField("field1").addRowField("nested", nestedSchema).build();
+
+    Schema expectedNestedSchema =
+        Schema.builder().addStringField("bottom1").addInt32Field("bottom2").build();
+    Schema expectedSchema =
+        Schema.builder()
+            .addStringField("top1")
+            .addRowField("top_nested", expectedNestedSchema)
```

Comment:
optional super nitty nit: I think this would be easier to grok if the nested schemas were defined inline.